extensions/datastores/kudu/src/main/java/org/locationtech/geowave/datastore/kudu/operations/KuduRangeRead.java

```diff
@@ -1,189 +1,266 @@
+                ComparisonOp.EQUAL,
+                partitionKey);
+        for (final ByteArrayRange range : r.getSortKeyRanges()) {
+          final byte[] start = range.getStart() != null ? range.getStart() : new byte[0];
+          final byte[] end =
+              range.getEnd() != null ? range.getEndAsNextPrefix()
+                  : new byte[] {
+                      (byte) 0xFF,
+                      (byte) 0xFF,
+                      (byte) 0xFF,
+                      (byte) 0xFF,
+                      (byte) 0xFF,
+                      (byte) 0xFF,
+                      (byte) 0xFF};
+          KuduPredicate lowerPred =
+              KuduPredicate.newComparisonPredicate(
+                  schema.getColumn(KuduField.GW_SORT_KEY.getFieldName()),
+                  ComparisonOp.GREATER_EQUAL,
+                  start);
+          KuduPredicate upperPred =
               KuduPredicate.newComparisonPredicate(
-                  schema.getColumn(partitionIdField),
-                  ComparisonOp.EQUAL,
-                  dataId);
-          KuduScannerBuilder scannerBuilder = operations.getScannerBuilder(table);
-          KuduScanner scanner =
-              scannerBuilder.addPredicate(adapterIdPred).addPredicate(partitionPred).build();
-          KuduUtils.executeQuery(scanner, results);
+                  schema.getColumn(KuduField.GW_SORT_KEY.getFieldName()),
+                  ComparisonOp.LESS,
+                  end);
+
+          AsyncKuduScannerBuilder scannerBuilder = operations.getAsyncScannerBuilder(table);
+          AsyncKuduScanner scanner =
+              scannerBuilder.addPredicate(lowerPred).addPredicate(upperPred).addPredicate(
+                  partitionPred).addPredicate(adapterIdPred).build();
+          scanners.add(scanner);
         }
-      } else {
-        KuduScannerBuilder scannerBuilder = operations.getScannerBuilder(table);
-        KuduScanner scanner = scannerBuilder.addPredicate(adapterIdPred).build();
-        KuduUtils.executeQuery(scanner, results);
       }
+    } else {
+      AsyncKuduScannerBuilder scannerBuilder = operations.getAsyncScannerBuilder(table);
+      AsyncKuduScanner scanner = scannerBuilder.addPredicate(adapterIdPred).build();
+      scanners.add(scanner);
     }
 
-    final Iterator<GeoWaveRow> tmpIterator;
-    Iterator<RowResult> concatIterator = Iterators.concat(results.iterator());
-    if (dataIds == null) {
-      Function<RowResult, GeoWaveRow> adapter;
-      if (isDataIndex) {
-        adapter = r -> KuduDataIndexRow.deserializeDataIndexRow(r, visibilityEnabled);
-      } else {
-        adapter = KuduRow::new;
+    return executeQueryAsync(scanners.toArray(new AsyncKuduScanner[] {}));
+  }
+
+  public CloseableIterator<T> executeQueryAsync(final AsyncKuduScanner... scanners) {
+    final BlockingQueue<Object> results = new LinkedBlockingQueue<>(MAX_BOUNDED_READS_ENQUEUED);
+    final AtomicBoolean isCanceled = new AtomicBoolean();
+    new Thread(new Runnable() {
```

Comment:
Could you replace this anonymous class with a lambda (() -> { <body of run()> })

Reply:
It says 'Syntax error on tokens, ClassInstanceCreationExpressionName expected instead' if I do that

Reply:
The code should look something like this:
```
    new Thread(() -> {
      final AtomicInteger queryCount = new AtomicInteger(1);
      ....
    }, "Kudu Query Executor").start();
```